tests/test_unit/approval_client/client.py

```diff
@@ -507,8 +507,10 @@ def _create_dataframe_info(dataframe: pd.DataFrame) -> dict[str, Any]:
                 "nan_count": int(dataframe.isna().sum().sum()),
                 "null_count": int(dataframe.isnull().sum().sum()),
                 "empty_count": int(dataframe[dataframe == ""].count().sum()),
-                "first_row": dataframe.iloc[0].round(4).to_dict(),
-                "last_row": dataframe.iloc[-1].round(4).to_dict(),
+                # We round float to 4 decimals places to avoid issues with floating point precision on different systems
+                # as this is stored a snapshot.
+                "first_row": dataframe.iloc[0].apply(lambda x: round(x, 4) if isinstance(x, float) else x).to_dict(),
+                "last_row": dataframe.iloc[-1].apply(lambda x: round(x, 4) if isinstance(x, float) else x).to_dict(),
```

Comment:
This is part of creating a snapshot. We use this for tests to ensure that nothing has changed in what is written to CDF from a set of configurations. In the newest version of `pandas` the `.round` is not supported on Series that are not of numeric dtype, thus instead we just do a check for each element instead.